sem_6/task_6_8/hw_6_3.py

```diff
@@ -0,0 +1,59 @@
+# Добавьте в пакет, созданный на семинаре шахматный модуль.
+# Внутри него напишите код, решающий задачу о 8 ферзях.
+# Известно, что на доске 8×8 можно расставить 8 ферзей так, чтобы они не били друг друга.
+# Вам дана расстановка 8 ферзей на доске, определите, есть ли среди них пара бьющих друг друга.
+# Программа получает на вход восемь пар чисел, каждое число от 1 до 8 - координаты 8 ферзей.
+# Если ферзи не бьют друг друга верните истину, а если бьют - ложь.
+import random
+
+
+def queens_beat(queens):
+    # Проверяем, разделяет ли какой-либо ферзь одну и ту же строку или столбец
+    for q1 in queens:
+        x1, y1 = q1
+        for q2 in queens:
+            x2, y2 = q2
+            if q1 != q2 and (x1 == x2 or y1 == y2):
+                return False
+
+    # Проверьте, имеет ли любой ферзь одну и ту же диагональ
+    for q1 in queens:
+        x1, y1 = q1
+        for q2 in queens:
+            x2, y2 = q2
+            if q1 != q2 and abs(x1 - x2) == abs(y1 - y2):
+                return False
+
+    return True
```

Comment:
Верно, ноу  вас два двухуровневых цикла. Сложность не самая низкая. Разберите такой вариант
```suggestion
    for i, (x1, y1) in enumerate(queens, 1):
        for x2, y2 in queens[i:]:
            if x1 == x2 or y1 == y2 or abs(x1 - x2) == abs(y1 - y2):
                return False

    return True
```